cohort-engine-api/src/main/java/com/ibm/cohort/engine/api/service/FHIRRestUtils.java

```diff
@@ -186,12 +198,45 @@ private static MeasureParameterInfo toMeasureParameterInfo(ParameterDefinition p
 		parameterDefinition.getExtension().stream()
 				.filter(MeasureEvaluationSeeder::isDefaultValue).findFirst() // only zero or one per IG
 				.map(Extension::getValue)
-				.map(Object::toString)
+				.map(x -> {if(defaultValue != null){return defaultValue;} else return x.toString();})
 				.ifPresent(retVal::setDefaultValue);
 
 		return retVal;
 	}
-	
+
+	static String complicatedTypeValueConstructor(ParameterDefinition parameterDefinition){
+		FhirContext context = FhirContext.forR4();
+		IParser parser = context.newJsonParser();
+		String retVal;
+		String intermediateValue;
+		Patient patient = new Patient();
+		Extension extension = new Extension();
+		switch (parameterDefinition.getType()){
+			case "Period":
+				Period period = (Period) parameterDefinition.getExtension().get(0).getValue();
+				extension.setValue(period);
+				patient.addExtension(extension);
+				break;
+			case "Range":
+				Range range = (Range) parameterDefinition.getExtension().get(0).getValue();
+				extension.setValue(range);
+				patient.addExtension(extension);
+				break;
+			case "Quantity":
+				Quantity quantity = (Quantity) parameterDefinition.getExtension().get(0).getValue();
+				extension.setValue(quantity);
+				patient.addExtension(extension);
+				break;
+			default:
+				throw new RuntimeException("Complicated Type" + parameterDefinition.getType() + " not yet implemented");
+		}
+		intermediateValue = parser.encodeResourceToString(patient);
+		intermediateValue = intermediateValue.replaceAll("\\{\"resourceType\":\"Patient\",\"extension\":\\[\\{\"url\":null,\"value.+?\":", "");
```

Comment:
Did you look at anything like JsonPath or the Jackson tree model for pulling the fields you care about out of the JSON?

I wouldn't expect the String representation to change out from under us, but if `extension` ever appears before `resourceType` down the line then this `replaceAll` will break.

Reply:
No, I didn't, I was under the impression that since the value name will be different for each type (valueRange valueQuantity) that I couldn't use something like that, I could always push that back into the switch statement. I'll try it out now.